javascript-typing-tutor/main.js

```diff
@@ -0,0 +1,24 @@
+var $letter = document.querySelectorAll('span');
+var counter = 0;
+
+function typingGame(event) {
+  if (event.key === $letter[counter].textContent) {
+    $letter[counter].className = 'green-letter';
+    if (counter < $letter.length - 1) {
+      $letter[counter + 1].className = 'underline';
+      counter++;
+    }
+  } else if (event.code === 'Space' && counter === 1) {
+    $letter[counter + 1].className = 'underline';
+    $letter[counter].className = '';
+    counter++;
+  } else if (event.code === 'Space' && counter === 6) {
```

Comment:
Rather than tying the spaces to hard-coded indexes, it'd probably be better to check if the content is a `&nbsp;`.